src/isl/isl_type.rs

```diff
@@ -159,53 +97,17 @@ impl NamedIslType {
                     ))
                 }
             };
-            // TODO: add more constraints to match below
-            let constraint = match constraint_name {
-                "all_of" => {
-                    //TODO: create a method/macro for this ion type check which can be reused
-                    if value.ion_type() != IonType::List {
-                        return Err(invalid_schema_error_raw(format!(
-                            "all_of constraint was a {:?} instead of a list",
-                            value.ion_type()
-                        )));
-                    }
-                    let types: Vec<IslTypeRef> = value
-                        .as_sequence()
-                        .unwrap()
-                        .iter()
-                        .map(|e| IslTypeRef::parse_from_ion_element(e))
-                        .collect::<IonSchemaResult<Vec<IslTypeRef>>>()?;
-                    IslConstraint::AllOf(types)
-                }
-                "type" => {
-                    if value.ion_type() != IonType::Symbol && value.ion_type() != IonType::Struct {
-                        return Err(invalid_schema_error_raw(format!(
-                            "type constraint was a {:?} instead of a symbol/struct",
-                            value.ion_type()
-                        )));
-                    }
-                    let type_reference: IslTypeRef = IslTypeRef::parse_from_ion_element(value)?;
-                    IslConstraint::Type(type_reference)
-                }
-                _ => {
-                    return Err(invalid_schema_error_raw(
-                        "Type: ".to_owned()
-                            + &type_name
-                            + " can not be built as constraint: "
-                            + constraint_name
-                            + " does not exist",
-                    ))
-                }
-            };
+
+            // set the isl type name for any error that is thrown while parsing its constraints
+            let mut isl_type_name = format!("{:?}", ion_struct);
+            if type_name.is_some() {
+                isl_type_name = type_name.to_owned().unwrap();
+            }
```

Comment:
This `format!` seems expensive. Why do anonymous types need a name? That is, why bother building a new string representation we won't use?

If there's a good reason, consider deferring the `format!` until you're sure it's necessary:
```suggestion
            let isl_type_name = match type_name {
                Some(name) => name,
                None => format!("{:?}", ion_struct)
            };
```